File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/SlimeBoss_Model.ts
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class SlimeBoss_Model extends MonsterInfo{
    public static Id = "SlimeBoss"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 9)
            return 150;
        else
            return 140;
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/boss/slime/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/boss/slime/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/boss/slime/skeleton.png"));
    }

    get Name(): string {
        return "怪物-史莱姆老大";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/SlimeBoss_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class SlimeBoss_Model : MonsterInfo
    {
        public static string Id = "SlimeBoss";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 9 ? 150 : 140;

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/boss/slime/skeleton.atlas",
                    "Monsters/theBottom/boss/slime/skeleton.json", "Monsters/theBottom/boss/slime/skeleton.png");
            }
        }

        public override string Name => "怪物-史莱姆老大";

        public SlimeBoss_Model(Vector2 pos) : base(pos)
        {
        }
    }
}